src/snakemake/persistence/__init__.py

```
@@ -0,0 +1,832 @@
+    def conda_cleanup_envs(self) -> None:
+        # cleanup envs
+        for address in set(
+            env.address
+            for env in self.dag.conda_envs.values()
+            if not env.is_externally_managed
+        ):
+            removed = False
+            if os.path.exists(address):
+                try:
+                    shutil.rmtree(address)
+                except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)